{"content":"\r\n  \r\n  <h1>\r\n    <span class=\"lang-csharp\">UpdateSourceExceptionFilterCallback</span>\r\n      Delegate\r\n  </h1>\r\n  \r\n  <h2 class=\"hiddenAnchor\">Definition</h2>\r\n  \r\n  <div class=\"metadata\">\r\n  \r\n  \r\n  </div>\r\n  \r\n  <div class=\"summaryHolder\">\r\n  \t<div class=\"summary\">\r\n  \t\t<p>Represents the method that handles exceptions that are thrown during the update of the binding source value. This must be used with the <a class=\"xref\" href=\"system.windows.controls.exceptionvalidationrule\" data-linktype=\"relative-path\">ExceptionValidationRule</a>.</p>\n\r\n  \t</div>\r\n  </div><nav id=\"center-doc-outline\" class=\"doc-outline\" ms.cmpgrp=\"intopic toc\" role=\"navigation\" aria-label=\"On page navigation\">\r\n    <h3>In this Article</h3>\r\n  </nav>\t<pre class=\"wrap loading\"><code class=\"csharp lang-csharp\">public delegate object UpdateSourceExceptionFilterCallback(object bindExpression, Exception exception);</code></pre>\r\n  \r\n  \r\n  \r\n  \r\n  \t<dl class=\"attributeList stack\">\r\n  \t\t<dt>Inheritance</dt><dd>\r\n  \t\t<div class=\"pdlHolder\">\r\n  \t\t<div class=\"pdl\"><span class=\"lang-csharp\"><a class=\"xref\" href=\"system.object\" data-linktype=\"relative-path\">Object</a></span></div><div class=\"pdl\"><span class=\"lang-csharp\"><a class=\"xref\" href=\"system.delegate\" data-linktype=\"relative-path\">System.Delegate</a></span></div><span class=\"lang-csharp\">UpdateSourceExceptionFilterCallback</span>\r\n  \t\t</div>\r\n  \t\t</dd>\r\n  \t</dl>\r\n  \r\n  \r\n  \r\n  \r\n  \r\n  \t<h2 class=\"accented\">Remarks</h2>\r\n  \t<p>If you have associated the <a class=\"xref\" href=\"system.windows.controls.exceptionvalidationrule\" data-linktype=\"relative-path\">ExceptionValidationRule</a> with your <a class=\"xref\" href=\"system.windows.data.binding\" data-linktype=\"relative-path\">Binding</a> object you have the option to use the <a class=\"xref\" href=\"system.windows.data.binding#System_Windows_Data_Binding_UpdateSourceExceptionFilter_\" data-linktype=\"relative-path\">UpdateSourceExceptionFilter</a> property to set this callback to provide custom logic for handling the exceptions. This callback is invoked whenever any exception is encountered when the binding engine updates the binding source value.  </p>\n<p> If an <a class=\"xref\" href=\"system.windows.data.binding#System_Windows_Data_Binding_UpdateSourceExceptionFilter_\" data-linktype=\"relative-path\">UpdateSourceExceptionFilter</a> is not specified on the <a class=\"xref\" href=\"system.windows.data.binding\" data-linktype=\"relative-path\">Binding</a>, the binding engine creates a <a class=\"xref\" href=\"system.windows.controls.validationerror\" data-linktype=\"relative-path\">ValidationError</a> with the exception and adds it to the <a class=\"xref\" href=\"https://msdn.microsoft.com/en-us/library/ms606099(v=vs.110).aspx\" data-linktype=\"external\">Validation.Errors</a> collection of the bound element.</p>\n\r\n  \r\n  \r\n  \r\n  \r\n  \r\n","outputRootRelativePath":"../","pageMetadata":"<meta name=\"author\" content=\"dotnet-bot\">\r\n<meta name=\"ms.author\" content=\"dotnetcontent\">\r\n<meta name=\"manager\" content=\"wpickett\">\r\n<meta name=\"breadcrumb_path\" content=\"/dotnet-all/api/toc2.json\">\r\n<meta name=\"ms.topic\" content=\"managed-reference\">\r\n<meta name=\"ms.prod\" content=\".net\">\r\n<meta name=\"search.ms_sitename\" content=\"Docs\">\r\n<meta name=\"search.ms_docsetname\" content=\"docs-internal\">\r\n<meta name=\"version\" content=\"0\">\r\n<meta name=\"locale\" content=\"en-us\">\r\n<meta name=\"site_name\" content=\"Docs\">\r\n<meta name=\"search.ms_product\" content=\"MSDN\">\r\n<meta name=\"depot_name\" content=\"MSDN.docs-internal\">\r\n<meta name=\"original_ref_skeleton_git_url\" content=\"https://github.com/dotnet/docs-internal/blob/master/api/System.Windows.Data.UpdateSourceExceptionFilterCallback.yml\">\r\n<meta name=\"APIName\" content=\"System.Windows.Data.UpdateSourceExceptionFilterCallback\">\r\n<meta name=\"APIName\" content=\"System.Windows.Data.UpdateSourceExceptionFilterCallback..ctor\">\r\n<meta name=\"APIName\" content=\"System.Windows.Data.UpdateSourceExceptionFilterCallback.Invoke\">\r\n<meta name=\"APIName\" content=\"System.Windows.Data.UpdateSourceExceptionFilterCallback.BeginInvoke\">\r\n<meta name=\"APIName\" content=\"System.Windows.Data.UpdateSourceExceptionFilterCallback.EndInvoke\">\r\n<meta name=\"TopicType\" content=\"apiref\">\r\n<meta name=\"APIType\" content=\"Assembly\">\r\n<meta name=\"updated_at\" content=\"2017-03-22 03:41 AM\">\r\n<meta name=\"document_id\" content=\"2046ad27-63f9-203d-122e-7d7d51c84dcc\">\r\n<meta name=\"pagetype\" content=\"Reference\">\r\n<meta name=\"description\" content=\"Represents the method that handles exceptions that are thrown during the update of the binding source value. This must be used with the .\n\">\r\n<meta name=\"toc_rel\" content=\"_splitted/system.windows.data/toc2.json\">\r\n<meta name=\"source_url\" content=\"\">\r\n<meta name=\"ms.assetid\" content=\"System.Windows.Data.UpdateSourceExceptionFilterCallback\">\r\n","rawMetadata":{"author":"dotnet-bot","ms.author":"dotnetcontent","manager":"wpickett","breadcrumb_path":"/dotnet-internal/toc.json","ms.topic":"managed-reference","ms.prod":".net","search.ms_sitename":"Docs","search.ms_docsetname":"docs-internal","version":0,"_op_canonicalUrlPrefix":"https://docs.microsoft.com/en-us/dotnet-internal/","locale":"en-us","site_name":"Docs","search.ms_product":"MSDN","_op_openToPublicContributors":false,"depot_name":"MSDN.docs-internal","_op_gitRefSkeletonCommitHistory":[],"original_ref_skeleton_git_url":"https://github.com/dotnet/docs-internal/blob/master/api/System.Windows.Data.UpdateSourceExceptionFilterCallback.yml","open_to_public_contributors":false,"api_name":["System.Windows.Data.UpdateSourceExceptionFilterCallback","System.Windows.Data.UpdateSourceExceptionFilterCallback..ctor","System.Windows.Data.UpdateSourceExceptionFilterCallback.Invoke","System.Windows.Data.UpdateSourceExceptionFilterCallback.BeginInvoke","System.Windows.Data.UpdateSourceExceptionFilterCallback.EndInvoke"],"api_location":[],"topic_type":["apiref"],"api_type":["Assembly"],"f1_keywords":["System.Windows.Data.UpdateSourceExceptionFilterCallback","System::Windows::Data::UpdateSourceExceptionFilterCallback"],"dev_langs":["csharp"],"updated_at":"2017-03-22 03:41 AM","document_id":"2046ad27-63f9-203d-122e-7d7d51c84dcc","content_git_url":"https://github.com/TianqiZhang/ECMA2YamlTestRepo2/blob/master/xml/System.Windows.Data/UpdateSourceExceptionFilterCallback.xml","layout":"Reference","_op_layout":"Reference","pagetype":"Reference","title":"UpdateSourceExceptionFilterCallback delegate | Microsoft Docs","_op_ogTitle":"UpdateSourceExceptionFilterCallback delegate","description":"Represents the method that handles exceptions that are thrown during the update of the binding source value. This must be used with the .\n","toc_asset_id":"api/_splitted/System.Windows.Data/toc.json","toc_rel":"_splitted/System.Windows.Data/toc.json","source_url":"","ms.assetid":"System.Windows.Data.UpdateSourceExceptionFilterCallback","canonical_url":"https://docs.microsoft.com/en-us/dotnet-all/api/system.windows.data.updatesourceexceptionfiltercallback","_op_searchScope":[".NET"],"_op_canonicalUrl":"https://docs.microsoft.com/en-us/dotnet-all/api/system.windows.data.updatesourceexceptionfiltercallback","fileRelativePath":"api/System.Windows.Data.UpdateSourceExceptionFilterCallback.html"},"themesRelativePathToOutputRoot":"_themes/"}